autotest/utils/evaluate_utils.py

```diff
@@ -0,0 +1,202 @@
+import csv
+import glob
+import os
+import subprocess
+
+from mmengine.config import Config
+
+DEFAULT_PORT = 23333
+
+
+def write_to_summary(model_name, tp_num, result, msg, worker_id, backend_type, work_dir=None):
+    status = '✅ PASS' if result else '❌ FAIL'
+
+    metrics = {}
+
+    if work_dir and os.path.exists(work_dir):
+        try:
+            summary_dirs = glob.glob(os.path.join(work_dir, '*', 'summary'))
+            if summary_dirs:
+                summary_dir = summary_dirs[0]
+                csv_files = glob.glob(os.path.join(summary_dir, 'summary_*.csv'))
+                if csv_files:
+                    csv_file = sorted(csv_files)[-1]
+                    if os.path.exists(csv_file):
+                        with open(csv_file, 'r') as f:
+                            reader = csv.reader(f)
+                            next(reader)
+                            for row in reader:
+                                if len(row) >= 5 and row[4]:
+                                    dataset = row[0]
+                                    metric_value = row[4]
+                                    try:
+                                        metrics[dataset] = f'{float(metric_value):.2f}'
+                                    except ValueError:
+                                        metrics[dataset] = metric_value
+        except Exception as e:
+            print(f'Error reading metrics: {str(e)}')
+
+    mmlu_value = metrics.get('mmlu', '')
+    gsm8k_value = metrics.get('gsm8k', '')
+
+    summary_line = f'| {model_name} | {backend_type} | TP{tp_num} | {status} | {mmlu_value} | {gsm8k_value} |\n'
+
+    summary_file = os.environ.get('GITHUB_STEP_SUMMARY', None)
+    if summary_file:
+        write_header = False
+        if not os.path.exists(summary_file) or os.path.getsize(summary_file) == 0:
+            write_header = True
+        else:
+            with open(summary_file, 'r') as f:
+                first_lines = f.read(200)
+                if '| Model | Backend | TP | Status | mmlu | gsm8k |' not in first_lines:
+                    write_header = True
```

Comment:
Can we change it to the following?
```
write_header = not os.path.exists(summary_file) or os.path.getsize(summary_file) == 0
```
May kindly let us know in what cases the first line are not the "| Model | Backend | TP | Status | mmlu | gsm8k |"